### PR TITLE
Zones2

### PR DESCRIPTION
**Pull request by joe-bowman**
## 1. Summary
Fixes QCK-285

## 2.Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## 3. Implementation details

- Remove validators from zones struct.
- Store validators as single KV entries
- Reduce number of gets/sets for each val

### CODE REVIEW

**Comment by aljo242** [comment]
@joe-bowman lint failing

File: x/participationrewards/keeper/rewards_validatorSelection.go [hunk 1]
```diff
@@ -84,23 +84,24 @@ func (k Keeper) CalcDistributionScores(ctx sdk.Context, zone icstypes.Zone, zs *
 	max := sdk.NewInt(0)
 	min := sdk.NewInt(999999999999999999)
 	for _, val := range zoneValidators {
-		if val.VotingPower.IsNegative() {
-			return fmt.Errorf("unexpected negative voting power for %s", val.ValoperAddress)
+		v := val
```
[the remark this answers is not in the document]
Comment (joe-bowman) [reply x/participationrewards/keeper/rewards_validatorSelection.go:87]: Ah -- two separate questions here :)

The assigning `v := val` I get; this makes sense. But `val := val` also works, right - it just changes the scope of `val`. My issue is with variables named `v` which should be more explicit.

[the remark this answers is not in the document]
Comment (aljo242) [reply x/participationrewards/keeper/rewards_validatorSelection.go]: Makes sense - fixed here: d45c31f93c9a91d121d6b247e3e157e58a046d6e

[the remark this answers is not in the document]
Comment (joe-bowman) [reply x/participationrewards/keeper/rewards_validatorSelection.go]: Thank you :)

**Comment by joe-bowman** [comment]
@aljo242 @ajansari95 - can we get this reviewed / approved please to upgrade testnet?

Thanks!

**Review by ajansari95 — APPROVED** [review]
LGTM

**Review by aljo242 — APPROVED** [review]

**Pull request merged**